command/base.go

```diff
@@ -76,6 +77,11 @@ type BaseCommand struct {
 // Client returns the HTTP API client. The client is cached on the command to
 // save performance on future calls.
 func (c *BaseCommand) Client() (*api.Client, error) {
+	if c.addrWarning != "" && c.UI != nil {
```

Comment:
This seems to break tests when the output is expected to be a particular format, e.g., JSON and the PKI HC tests.

Reply:
@cipherboy I am thinking if it may be an idea to adjust the tests to explicitly set an address which would the least alteration.
https://github.com/hashicorp/vault/blob/main/command/pki_verify_sign_test.go#L99-L113

Reply:
I think the point still holds: why are we injecting non-JSON formatted output without modifying the ret code? This seems like a change that can break shell scripts :-) 

(FWIW, I think the same point holds with the `-field` parameter as well).

Definitely, if it was limited to interactive usage (or, modes which are unlikely to be used in shell scripts), I'd be all for the change as it seems like a great improvement!

Reply:
Seems to be fixed with the latest suggestion from @ncabatoff - all tests are passing now.

Reply:
I'm a little antsy about the same point @cipherboy brings up here, the risk of breaking shell scripts.  Perhaps it's worth that risk, but I want to think about it a bit more before approving.